Feature/product details

lemongucci opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
- Product details 페이지 프레임 작업 및 각 section 컴포넌트화
		 
## 수정 사항들 자세한 내용
- section1 Description 을 2개의 컴포넌트로 구성. LeftSlider, RightSlider.
- LeftSlider는 현재 parallax scrolling 작업 중
- LeftSlider 완성 후, LeftSlider와 RightSlider의 scrolling sync 작업할 예정
- section3 ReviewList 컴포넌트화

## 기타 질문 및 특이 사항
- Parallax scrolling 작업을 Javascript를 이용해서 작업 중 관련된 package나 library를 찾아서 사용해볼 지 말지 고민 중 

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [X] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [X] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [X] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [X] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

Commit:
Add: ProductDetails 페이지 구성 파일 및 Routes 파일 생성
- 2020년 10월 20일 향후 팀원 각자 작업한 컴포넌트를 UI에 원활히 띄우기 위해 각 페이지에 해당되는 path를 컴포넌트명과 동일하게 지정
- 개인 작업 ProductDetails는 section 3개로 구분하였다. section 2에 해당되는 YouMayAlsoLike는 벡엔드와 상의한 후 기존 데이터에서 랜덤하게 제시하기로 결정.

Commit:
Merge branch 'main' of https://github.com/wecode-bootcamp-korea/13-getSuperCool-frontend into main

saengmotmi commented:
@lemongucci 리뷰 요청 라벨 달아주세요

Commit:
Add: RightSlider의 레이아웃 작업을 위한 태그 추가
- LeftSlider의 애니메이션 효과는 Parallax scrolling으로 구현하기로 결정
- RightSlider 레이아웃 작업 중, 백엔드의 데이터 모델링을 바탕으로 Mockdata를 이용해 기본 기능 구현할 계획

Commit:
Add: Section1 LeftSlider의 이미지를 감싸는 div 태그 추가
- ProductDetails Section1의 LeftSlider, RightSlider의 레이아웃 초기 작업완료
- LeftSlider와 RightSlider의 Scrolling을 조절할 예정
- RightSlider는 Pagination을 통해 구현할 지, 아니면 state, props를 사용해서 부모, 자식 컴포넌트 개념으로 구현할 지 고민 중

claire-euni commented on src/Pages/ProductDetails/ComponentYH/LeftSlider.js:
```diff
@@ -0,0 +1,53 @@
+/* eslint-disable jsx-a11y/alt-text */
```
주석 부분은 삭제해주세요~

claire-euni commented on src/Pages/ProductDetails/ComponentYH/LeftSlider.js:
```diff
@@ -0,0 +1,53 @@
+      hah:"fdf"
+    };
+  }
+
+
```
간격은 엔터 한줄이면 충분합니다

claire-euni commented on src/Pages/ProductDetails/ComponentYH/LeftSlider.js:
```diff
@@ -0,0 +1,53 @@
+          <img src='https://images.ctfassets.net/vnxry7jc7f2k/6WYylPT5lY1WCC9dDi51xK/bfcd283d6d647370a9754b38aa71b25d/resize-09.jpg?w=1440&h=1800&q=80&fm=webp'></img>
+        </div>
+        <div className='Foreground'>
+          <div className='Fore Icon1'>
+            <img className='Icon1' src='https://i.ibb.co/wS6dvTj/icontrans.png'></img>
+          </div>
```
반복되는 코드부분은 map()을 활용해보세요 :)

claire-euni commented on src/Pages/ProductDetails/ComponentYH/LeftSlider.js:
```diff
@@ -0,0 +1,53 @@
+  constructor(){
+    super();
+    this.back = React.createRef();
+    this.state = {
+      hah:"fdf"
```
hah .........?

claire-euni commented on src/Pages/ProductDetails/ComponentYH/LeftSlider.js:
```diff
@@ -0,0 +1,53 @@
+  }
+
+
+  BackgroundScroll = (e) => {
+    if(e.target.clientHeight === window.outerHeight){
+      this.back.current.style.overflow = "scroll"
+    }
+  }
```
- window.outerHeight을 필요로 하는 함수인가요? 어떤 기능을 하는 함수인가요?
- 함수의 이름은 소문자로 시작하게 해주세요

claire-euni commented on src/Pages/ProductDetails/ComponentYH/LeftSlider.js:
```diff
@@ -0,0 +1,53 @@
+  render(){
+    const Back = this.back.current;
+    return (
+      <div className='Test'>
+        <div className='Background'  ref={this.back}>
```
- 컴포넌트의 최상위 태그명은 컴포넌트와 동일하게 대문자로 시작하게 지어주세요. `<div className="LeftSlider>`
- Background 클래스를 가진 태그는 className과 ref의 간격이 머네요...스페이스 한칸이면 충분합니다 :)

claire-euni commented on src/Pages/ProductDetails/ComponentYH/LeftSlider.scss:
```diff
@@ -0,0 +1,303 @@
+    position: relative;
+
+    .Background {
+        display: flex;
+        flex-direction: column;
+        position: absolute;
```
자주 쓰이는 css속성이라면 @mixin을 이용해보면 좋을 것 같습니다 🤗

claire-euni commented on src/Pages/ProductDetails/ComponentYH/LeftSlider.scss:
```diff
@@ -0,0 +1,303 @@
+            width: 50vw;
+            height: 85vw;
+        }
+        
+
```
셀렉터 사이 간격도 엔터 한줄만 주세요~

claire-euni commented on src/Pages/ProductDetails/ComponentYH/LeftSlider.scss:
```diff
@@ -0,0 +1,303 @@
+        background-color: transparent;
+        z-index: 20;
+
+        .Fore.Icon1 {
+            width: 50vw;
+            height: 85vw;
+        }
+        
+
+        .Fore {
+            width: 50vw;
+            height: 100vh;
+            object-fit: cover;
+            background-color: transparent;
+            position: relative;
```
width, height이 중복적용되고 있으니 최상위에 하나만 남겨주시고 자식 태그에서는 속성을 삭제해주세요

claire-euni commented on src/Pages/ProductDetails/ComponentYH/LeftSlider.scss:
```diff
@@ -0,0 +1,303 @@
+
+//     .title{
+//         font-size: 1.2rem;
+//     }
```
불필요한 엔터들과 주석처리 부분은 삭제해주세요

claire-euni commented on src/Pages/ProductDetails/ComponentYH/ReviewList.js:
```diff
@@ -0,0 +1,14 @@
+// import { render } from "node-sass";
```
주석부분 삭제해주세요

claire-euni commented on src/Pages/ProductDetails/ComponentYH/RightSlider.js:
```diff
@@ -0,0 +1,132 @@
+            <li>notes&ingredients</li>
+            <li>shipping&returns</li>
+          </ul>
+          <p className='description'>A true multi-tasker.
+
+On-the-go two-in-one stick for lips and cheeks that blends in seamlessly for an instant flushed effect, with zero effort.
+
+Buildable and weightless cream-to-powder formula ready to give us a pearlescent pop of color!</p>
```
불필요한 엔터들은 삭제해주세요

claire-euni commented on src/Pages/ProductDetails/ComponentYH/RightSlider.js:
```diff
@@ -0,0 +1,132 @@
+            <p>Awesome!!! Subscribe!!! To News Letter!!!</p>
+          </div>
+          <div className='Philosophy'>
+            <div className='Humane'>
+              <img alt='Humane'src='https://i.ibb.co/0X07wRq/Image-from-i-OS-2.jpg'></img>
+              <p>Humane</p>
+            </div>
```
이 부분도 map()을 활용하면 코드를 간결하게 만들 수 있을 것 같습니다 :)

claire-euni commented on src/Pages/ProductDetails/ComponentYH/RightSlider.js:
```diff
@@ -0,0 +1,132 @@
+  }
+}
+
+const MAPPING_OBJ = {
+  // 1: <First/>,
+  // 2: <Second/>,
+  // 3: <Third/>,
+}
+
+const MAPPING_MENU = ["First", "Second", "Third"];
+
+export default RightSlider;
+{/* <div className='wrapper'>
+  <ul className='tabs'>
+    {MAPPING_MENU.map(
+      (menu, idx) => (
+        <li onClick={()=> this.OpenPage(idx+1)}>{menu}</li>
+      )
+    )}
+  </ul>
+  <div className='Contents'>
+    {MAPPING_OBJ[activeId]}
+  </div>
+</div> */}
```
클래스 외에서 변수를 선언한다면 일반적으로 클래스 위에서 설정하는것이 컨벤션입니다~ 
주석처리 부분 모두 삭제해주세요 ~

claire-euni commented on src/Pages/ProductDetails/ComponentYH/RightSlider.scss:
```diff
@@ -0,0 +1,171 @@
+    width: 50vw;
+    height: 80vw;
+    
+    .ProductName {
+        h1{
+            text-transform: uppercase;
+            font-size: 3.13vw;
+            font-weight: bold;
+        }
+
+        span {
+            text-transform: uppercase;
+            font-weight: bold;
+            font-size: 1.74vw;
+        }
+    }
```
공통으로 적용할 속성이라면 상위에서 한번만 주면 되겠죠? 
```suggestion
    .ProductName {
            text-transform: uppercase;
            font-weight: bold;
            
            h1 {
                font-size: 3.13vw;
            }
            
            span {
                font-size:1.74vw;
            }
}
```

claire-euni commented on src/Pages/ProductDetails/ComponentYH/RightSlider.scss:
```diff
@@ -0,0 +1,171 @@
+        }
+
+        p.description {
+            font-size: 0.69vw;
+        }
+
+        p.ingredients {
+            font-size: 0.69vw;
+        }
+
+        p.shipping {
+            font-size: 0.69vw;
```
동일하게 적용된다면 p 태그에 `font-size:0.69vw;` 를 주시면 될 것 같습니다

claire-euni commented on src/Pages/ProductDetails/ComponentYH/RightSlider.scss:
```diff
@@ -0,0 +1,171 @@
+            width: 18.06vw;
+            height: 4.24vw;
+            display: flex;
+            flex-direction: row;
```
`display:flex;`의 direction이 어느 방향이죠? 
필요한 속성일까요?

claire-euni commented on src/Pages/ProductDetails/ComponentYH/RightSlider.scss:
```diff
@@ -0,0 +1,171 @@
+
+        div {
+            display: flex;
+            flex-direction: row;
```
이 부분도 마찬가지로 필요할까요?

claire-euni commented on src/Pages/ProductDetails/ComponentYH/RightSlider.scss:
```diff
@@ -0,0 +1,171 @@
+            justify-content: space-between;
+
+            .Adding{
+                display: flex;
+                justify-content: space-between;
+                align-items: center;
```
자주 쓰이는 속성이라면 @mixin!

claire-euni commented on src/Pages/ProductDetails/ProductDetails.js:
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+// import Logo from "../logo_text.png";
```
주석부분은 삭제해주세요

claire-euni commented on src/Pages/ProductDetails/ProductDetails.js:
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+// import Logo from "../logo_text.png";
+import "./ProductDetails.scss";
+
+import LeftSlider from "./ComponentYH/LeftSlider";
+import RightSlider from "./ComponentYH/RightSlider";
+import ReviewList from "./ComponentYH/ReviewList";
```
import사이에는 엔터간격을 주지 않아도 괜찮습니다. 
import의 순서는 일반적으로 React -> Library -> Component -> Style 순서입니다. 수정해주세요 :)

claire-euni commented on src/Routes.js:
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import {Switch, Route} from "react-router-dom";
+
```
불필요한 엔터간격입니다 ~

claire-euni commented on src/index.js:
```diff
@@ -1,17 +1,24 @@
+  // <LeftSlider/>,
+  // <RightSlider/>,
+
+  <ProductDetails/>,
```
ProductDetail 컴포넌트내에 LeftSlider 와 RightSlider가 들어있으므로 index.js에서 또 import 해줄 필요는 없습니다

claire-euni reviewed (commented):
리뷰반영해서 리팩토링 진행해주세요 양효님 화이팅🌼

Commit:
Add: 테스트용 Mock data 추가
- 2020년 10월 26일 ProductDetails Section1 스타일링 작업 완료
- ProductDetails의 애니메이션, 데이터 흐름을 고려한 효율적인 컴포넌트 사용 계획 완료
- 백엔드 데이터 구조에 맞는 RightSlider 기능 구현은 수요일까지 완료할 예정
- Section2, Section3 작업 시작
- 현재 ProductDeatils 애니메이션 기능 구현 작업 중

Commit:
Fix: git 충돌 해결
- common.scss, index.js 파일 수정

Commit:
Merge branch 'main' into feature/ProductDetails

Commit:
Add: RightSlider Add to Cart 부분 기능
- 버튼을 클릭하여 숫자의 증감에 따라 총 가격이 변경되어 totPrice 상태에 저장되게 하였다.
- 나중에 백엔드 쪽으로 상품 아이디, 사용자 정보, 총 가격을 넘어가게 만들 예정
